tests/ui/uninit_vec.rs

```diff
@@ -0,0 +1,32 @@
+#![warn(clippy::uninit_vec)]
+
+use std::mem::MaybeUninit;
+
+fn main() {
+    // with_capacity() -> set_len() should be detected
+    let mut vec: Vec<u8> = Vec::with_capacity(1000);
+    unsafe {
+        vec.set_len(200);
+    }
+
+    // reserve() -> set_len() should be detected
+    vec.reserve(1000);
+    unsafe {
+        vec.set_len(200);
+    }
+
+    // test when both calls are enclosed in the same unsafe block
+    unsafe {
+        let mut vec: Vec<u8> = Vec::with_capacity(1000);
+        vec.set_len(200);
+
+        vec.reserve(1000);
+        vec.set_len(200);
+    }
+
+    // MaybeUninit-wrapped types should not be detected
+    let mut vec: Vec<MaybeUninit<u8>> = Vec::with_capacity(1000);
```

Comment:
Can you also add tests for the other types accepted by `is_uninit_ty_valid`? This would be an array of `MaybeUninit<T>` and a tuple of them :)